src/traverse.rs

```diff
@@ -3,24 +3,48 @@ use anyhow::Result;
 use filesize::PathExt;
 use petgraph::{graph::NodeIndex, stable_graph::StableGraph, Directed, Direction};
 use std::{
+    fmt,
     fs::Metadata,
     io,
     path::{Path, PathBuf},
-    time::Duration,
+    time::{Duration, SystemTime, UNIX_EPOCH},
 };
 
 pub type TreeIndex = NodeIndex;
 pub type Tree = StableGraph<EntryData, (), Directed>;
 
-#[derive(Eq, PartialEq, Debug, Default, Clone)]
+#[derive(Eq, PartialEq, Clone)]
 pub struct EntryData {
     pub name: PathBuf,
     /// The entry's size in bytes. If it's a directory, the size is the aggregated file size of all children
     pub size: u128,
+    pub mtime: SystemTime,
```

Comment:
Could we have a test along the lines of `assert_eq!(std::mem::size_of::<EntryData>(), X)` to see what the size of this is now?

You could just rebase onto `main` to get such a test, I added it. I'd not expect this to have huge impact, but I'd like to be sure. Probably it's best to run both versions, with and without mtime, on a big tree and compare the consumption for completeness.

Reply:
I have just rebased, and re-runt the tests:

```
---- traverse::tests::size_of_entry_data stdout ----
thread 'traverse::tests::size_of_entry_data' panicked at 'assertion failed: `(left == right)`
  left: `64`,
 right: `48`: the size of this should not change unexpectedly as it affects overall memory consumption', src/traverse.rs:267:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Looks like mtime adds 16bytes per entry.

Reply:
Let me know if this is too much in your opinion, and I will replace it with a single u64 for unix timestamp instead (8 bytes).

Reply:
Yes, that's a good idea. Let's see if a single u64 would actually safe 8 bytes, but if so I think we should have it. Nanosecond precision for mtimes isn't needed, after all.

> [!NOTE]
> If you are inclined to have nanoseconds, you can also use an Ext4-style encoding of 30 bits of nanoseconds and 32 bits of seconds, along with 2 bits of epoch. But that's entirely optional and to my mind, unnecessary (*I just like the encoding and wanted to have mentioned it 😁*)

Reply:
I have made the change, but it has not saved any memory, because of struct memory alignment. The largest member in `EntryData` is 8 bytes (`size` field), so the other fields are aligned to the same size (https://levelup.gitconnected.com/how-struct-memory-alignment-works-in-c-3ee897697236).

What do you think? Given the above problem, using `SystemTime` actually makes the code cleaner, as otherwise I have to deal with the problem of converting timestamp back to a `DateTime` and formatting it.

Reply:
I had a hunch this could happen as well. Let's keep it then.